src/main/java/de/tum/cit/aet/artemis/core/web/admin/AdminCleanupResource.java

```diff
@@ -0,0 +1,112 @@
+package de.tum.cit.aet.artemis.core.web.admin;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.core.dto.CleanupServiceExecutionRecordDTO;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAdmin;
+import de.tum.cit.aet.artemis.core.service.cleanup.DataCleanupService;
+
+/**
+ * REST controller for managing old data cleanup operations in Artemis.
+ * Provides endpoints for administrators to clean up old or orphaned data in the database.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/admin/cleanup/")
+@EnforceAdmin
+public class AdminCleanupResource {
+
+    private static final Logger log = LoggerFactory.getLogger(AdminCleanupResource.class);
+
+    private final DataCleanupService dataCleanupService;
+
+    public AdminCleanupResource(DataCleanupService dataCleanupService) {
+        this.dataCleanupService = dataCleanupService;
+    }
+
+    /**
+     * DELETE admin/cleanup/delete-orphans
+     * Deletes orphaned data in the Artemis database.
+     *
+     * @return a {@link ResponseEntity} containing the result of the cleanup operation
+     */
+    @DeleteMapping("delete-orphans")
+    public ResponseEntity<CleanupServiceExecutionRecordDTO> deleteOrphans() {
+        log.debug("REST request to delete orphaned data in Artemis database");
+        CleanupServiceExecutionRecordDTO result = dataCleanupService.deleteOrphans();
+        return ResponseEntity.ok().body(result);
+    }
+
+    /**
+     * DELETE admin/cleanup/delete-plagiarism-comparisons
+     * Deletes plagiarism comparisons within the specified date range.
+     *
+     * @param deleteFrom the start date of the deletion range
+     * @param deleteTo   the end date of the deletion range
+     * @return a {@link ResponseEntity} containing the result of the cleanup operation
+     */
+    @DeleteMapping("delete-plagiarism-comparisons")
```

Comment:
Something I'm noticing is that the HTTP Method (DELETE) is reflected twice in these endpoints. I remember [a comment](https://github.com/ls1intum/Artemis/pull/9296#pullrequestreview-2334081358) from Stephan that the URL should not contain the method name. Maybe `api/admin/cleanup/plagiarism-comparisons`?

Also applies to other endpoints in this file